samza-core/src/main/java/org/apache/samza/operators/StreamGraphSpec.java

```diff
@@ -91,15 +86,15 @@ public void setDefaultSerde(Serde<?> serde) {
 
   @Override
   public <M> MessageStream<M> getInputStream(String streamId, Serde<M> serde) {
-    StreamSpec streamSpec = runner.getStreamSpec(streamId);
```

Comment:
Just one thought: if the reason to remove the reference to StreamSpec is simply because it is accessed via runner.getStreamSpec() method, can we replace it with StreamSpecManager.getStreamSpec() here? It seems that here the only reason that we are calling runner.getStreamSpec() is because we want to construct the StreamSpec from the configuration, which can be put into StreamSpecManager class.

Reply:
Make sense!